assets/js/components/notifications/SetupSuccessBannerNotification.js

```diff
@@ -95,32 +91,67 @@ function SetupSuccessBannerNotification() {
 			)
 	);
 
-	useMount( () => {
-		trackEvent(
-			`${ viewContext }_authentication-success-notification`,
-			'view_notification'
-		);
-
-		// Only trigger these events if this is a site/plugin setup event,
-		// and not setup of an individual module (eg. AdSense, Analytics, etc.)
-		if ( slug === null ) {
-			trackEvent(
-				`${ viewContext }_authentication-success-notification`,
-				'complete_user_setup',
-				isUsingProxy ? 'proxy' : 'custom-oauth'
-			);
+	const [ viewNotificationSent, setViewNotificationSent ] = useState( false );
+	const [ completeUserSetupSent, setCompleteUserSetupSent ] =
+		useState( false );
+	const [ completeSiteSetup, setCompleteSiteSetup ] = useState( false );
 
-			// If the site doesn't yet have multiple admins, this is the initial
-			// site setup so we can log the "site setup complete" event.
-			if ( ! hasMultipleAdmins ) {
+	useEffect( () => {
+		// Only trigger the GA events if the notification is visible and we haven't
+		// already sent these notifications.
+		if (
+			modules !== undefined &&
+			notification === 'authentication_success'
+		) {
+			if (
+				! viewNotificationSent &&
+				canManageOptions &&
+				slug &&
+				modules[ slug ]?.active
+			) {
 				trackEvent(
 					`${ viewContext }_authentication-success-notification`,
-					'complete_site_setup',
-					isUsingProxy ? 'proxy' : 'custom-oauth'
+					'view_notification'
 				);
+				setViewNotificationSent( true );
+			}
+
+			// Only trigger these events if this is a site/plugin setup event,
+			// and not setup of an individual module (eg. AdSense, Analytics, etc.)
+			if ( slug === null ) {
+				if ( ! completeUserSetupSent ) {
+					trackEvent(
+						`${ viewContext }_authentication-success-notification`,
+						'complete_user_setup',
+						isUsingProxy ? 'proxy' : 'custom-oauth'
+					);
+					setCompleteUserSetupSent( true );
+				}
+
+				// If the site doesn't yet have multiple admins, this is the initial
+				// site setup so we can log the "site setup complete" event.
+				if ( ! completeSiteSetup && hasMultipleAdmins === false ) {
+					trackEvent(
+						`${ viewContext }_authentication-success-notification`,
+						'complete_site_setup',
+						isUsingProxy ? 'proxy' : 'custom-oauth'
+					);
+					setCompleteSiteSetup( true );
+				}
 			}
```

Comment:
I think we can reduce the nesting here too to improve readability:

```suggestion
			// Only trigger these events if this is a site/plugin setup event,
			// and not setup of an individual module (eg. AdSense, Analytics, etc.)
			if ( slug === null && ! completeUserSetupSent ) {
				trackEvent(
					`${ viewContext }_authentication-success-notification`,
					'complete_user_setup',
					isUsingProxy ? 'proxy' : 'custom-oauth'
				);
				
				setCompleteUserSetupSent( true );
			}

			// If the site doesn't yet have multiple admins, this is the initial
			// site setup so we can log the "site setup complete" event.
			if ( slug === null && ! completeSiteSetup && hasMultipleAdmins === false ) {
				trackEvent(
					`${ viewContext }_authentication-success-notification`,
					'complete_site_setup',
					isUsingProxy ? 'proxy' : 'custom-oauth'
				);

				setCompleteSiteSetup( true );
```